DynamicAnalyzer/views/android/environment.py

```diff
@@ -95,17 +113,26 @@ def configure_proxy(self, project):
 
     def install_mobsf_ca(self, action):
         """Install or Remove MobSF Root CA."""
+        ca_construct = '{}.0'
+        ca_file_hash = subprocess.check_output(['openssl',
```

Comment:
Could you please explain why this is required?

Reply:
The mitmproxy script is generating a CA cert at first start. The conversation to store the certificate in the Android certificate store is to use a part of the hash. this is build with the constuctor in the code and the has, identified with openssl. (originally it was hardcoded in the config)

Reply:
So does the name of cert matters to Android to identify the user installed certificate?

If so, there needs to be an alternative to openssl. This won't work in default windows setup.

Reply:
I understand the problem on windows - but I didn't get it as I am running on linux / Ubuntu Docker. Additionally, I cannot say if this is required, but this is the way normally be conform with the storage of a CA certificate file. You have to decide, as it is just a suggestion. Sorry that I have no further information of how to handle it on windows.

Reply:
for openssl you should use shutil.which() to find the full path of openssl and if openssl is available in the path and raise an error if not

Reply:
We can't have it in code if it's not cross platform without external dependencies. If the naming doesn't affect the functionality, I suggest to keep it hardcoded like before.

Or may be something like

```
>>> import os
>>> from OpenSSL import crypto
>>> pem = open(os.path.expanduser('~/.mitmproxy/mitmproxy-ca-cert.pem'), 'rb').read()
>>> cert=crypto.load_certificate(crypto.FILETYPE_PEM, pem)
>>> cert.subject_name_hash()
2344488589
```

Not sure if it's the same as `subject_hash_old`

Reply:
Thank you for the good alternative solution. I am trying to apply your suggestion (as it is not the same as in openssl).
Background information, why I think it would be useful (for performance):

OpenSSL has implemented a filename technique to reduce the Certificate Authority (CA) look-up time. Without this approach it would be required to read all CA files in the CApath to find the matching certificate.

Reply:
The only thing missing was the conversion to hex. Now it is using the newer SHA1 hash than the "old" MD5 one. Should be fine, as it is tested in my setup. Please review.